tests/conftest.py

```diff
@@ -122,3 +130,28 @@ def constructor_eager(
 @pytest.fixture(params=[*eager_constructors, *lazy_constructors])
 def constructor(request: pytest.FixtureRequest) -> Constructor:
     return request.param  # type: ignore[no-any-return]
+
+
+@pytest.fixture
```

Comment:
should these be `@pytest.fixture(scope="session")`? then they won't get recalculated for each test, and maybe `hypothesis` won't complain about them being used?

Reply:
right, thanks! :)
Some tests on old versions are still failing, but I suspect this might be due to my improper use of `request.applymarker(pytest.mark.skip)` (see eg https://github.com/pytest-dev/pytest/issues/3969). I'm exploring in this direction

Reply:
i think most of them can probably just be `request.applymarker(pytest.mark.xfail)`?

Reply:
agree! oor, as most of them were originally skipped, keep on skipping them (via explicit `pytest.skip` though) as this will happen inside conditions checking for `PANDAS_VERSION`, `POLARS_VERSION` etc. What do you think?

Reply:
sure, ok to keep skipping them, perhaps the constants can be useful here then? like
```
@pytest.mark.skipif(
    PANDAS_VERSION < (2,), reason="convert_dtypes not available"
)
```